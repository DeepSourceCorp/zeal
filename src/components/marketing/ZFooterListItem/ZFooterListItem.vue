<template>
  <li
    class="flex items-center pt-2 pb-1 text-sm whitespace-nowrap"
    :class="[`${hoverColor && `hover:text-${hoverColor}`}`, `${cursorPointer && 'cursor-pointer'}`]"
  >
    <slot></slot>
  </li>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  name: 'ZNavListItem',
  props: {
    hoverColor: {
      type: String,
      default: 'juniper'
    },
    cursorPointer: {
      type: Boolean,
      default: true
    }
  }
})
</script>
