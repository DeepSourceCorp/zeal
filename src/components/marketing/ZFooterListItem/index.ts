import ZFooterListItem from './ZFooterListItem.vue'

export default ZFooterListItem