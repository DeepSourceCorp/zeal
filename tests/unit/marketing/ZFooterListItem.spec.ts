import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZFooterListItem from '../../../src/components/marketing/ZFooterListItem'

describe('ZFooterListItem component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZFooterListItem, {
        localVue,
        ...options
      })
    }
  })
  it('renders a ZFooterListItem when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })
})
