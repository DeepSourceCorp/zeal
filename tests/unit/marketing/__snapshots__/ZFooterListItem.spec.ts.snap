// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZFooterListItem component renders a ZFooterListItem when the component is mounted 1`] = `<li class="flex items-center pt-2 pb-1 text-sm whitespace-nowrap hover:text-juniper cursor-pointer"></li>`;
